### PR TITLE
[PWPA-1937] Convert annuli data for the APB plugin

### PR DESCRIPTION
**Pull request by gabrielantao**
In this PR I created more readers for some needed properties for annuli and I created the class `ScoreAPBPluginConverter` exclusively to build the plugin data.

PWPA-1937

### CODE REVIEW

File: src/alfasim_score/converter/alfacase/score_input_reader.py [hunk 1]
```diff
@@ -338,6 +359,37 @@ def read_operation_fluid_data(self) -> Dict[str, Any]:
...
+                {
+                    "string_name": annulus["string_name"],
+                    "leakoff_volume": Scalar(leakoff_volume, VOLUME_UNIT),
+                    "initial_top_pressure": Scalar(annulus["initial_top_pressure"], PRESSURE_UNIT),
+                }
+            )
+        return annuli_data
+
+    def read_tubing_fluid_data(self) -> List[Dict[str, Any]]:
+        """Read data for the fluid associated to tubing in SCORE input file."""
```
Comment (carlosnewmar) [inline src/alfasim_score/converter/alfacase/score_input_reader.py]: """Read data for the annulus fluid associated to tubing in SCORE input file."""

File: src/alfasim_score/converter/alfacase/score_input_reader.py [hunk 2]
```diff
@@ -338,6 +359,37 @@ def read_operation_fluid_data(self) -> Dict[str, Any]:
             "gas_gravity": Scalar(fluid_data["gas_gravity"], FRACTION_UNIT),
         }
 
+    def read_operation_annuli_data(self) -> List[Dict[str, Any]]:
+        """Read data for the annuli for the operation registered in SCORE input file."""
```
Comment (carlosnewmar) [inline src/alfasim_score/converter/alfacase/score_input_reader.py]: """Read annuli data defined in operation in SCORE input file."""

File: src/alfasim_score/converter/alfacase/score_input_reader.py [hunk 3]
```diff
@@ -177,6 +179,25 @@ def read_casings(self) -> List[Dict[str, Any]]:
                         "final_md": Scalar(item["final_md"], LENGTH_UNIT, "length"),
```
Comment (carlosnewmar) [inline src/alfasim_score/converter/alfacase/score_input_reader.py:179]: """Read geometric data for the casing and annulus data associated to the casing from SCORE input file."""

File: src/alfasim_score/conftest.py [hunk 4]
```diff
@@ -1,6 +1,20 @@
+from typing import Any
```
Comment (ckiyono) [inline src/alfasim_score/conftest.py]: Are all these imports needed?

**Review by carlosnewmar — APPROVED** [review]

**Review by ckiyono — APPROVED** [review]
👍

File: src/alfasim_score/common.py [hunk 5]
```diff
@@ -58,6 +67,114 @@ def _missing_(cls, value):  # type: ignore
...
+            "is_active": "is_active",
+            "mode_type": "mode_type",
+            "initial_top_pressure": "initial_top_pressure",
+            "is_open_seabed": "open_sea",
+            "annulus_table": "annulus_table",
+            "has_fluid_return": "fluid_return",
+            "initial_leakoff": "initial_leakoff",
+            "has_relief_pressure": "relief_pressure_check",
+            "relief_pressure": "pressure_relief",
+            "relief_position": "relief_position",
```
Comment (noliveiraesss) [inline src/alfasim_score/common.py:147]: Some things must change here, since I've followed your suggestions on PWPA-2113.

Comment (noliveiraesss) [reply src/alfasim_score/common.py:147]: Also add the new temperature vs depth table.

File: src/alfasim_score/converter/alfacase/_tests/test_build_injection_alfacase/test_create_alfacase_injection.alfacase [hunk 6]
```diff
@@ -82,7 +82,301 @@ numerical_options:
   caching_atol: 0.0001
   always_repeat_timestep: False
   enable_fast_compositional: True
-plugins: []
+plugins:
+- name: apb
```
Comment (noliveiraesss) [inline src/alfasim_score/converter/alfacase/_tests/test_build_injection_alfacase/test_create_alfacase_injection.alfacase:86]: The same for all .alfacase

**Review by noliveiraesss — APPROVED** [review]
Great job!
I've made some comments about changes, but it's up to you to do it now or in a future PR.

File: src/alfasim_score/common.py (hunk 5 above)
Comment (gabrielantao) [reply src/alfasim_score/common.py:147]: ok, done in the next PR.

File: src/alfasim_score/converter/alfacase/_tests/test_build_injection_alfacase/test_create_alfacase_injection.alfacase (hunk 6 above)
Comment (gabrielantao) [reply src/alfasim_score/converter/alfacase/_tests/test_build_injection_alfacase/test_create_alfacase_injection.alfacase:86]: Done in next PR.

File: src/alfasim_score/conftest.py (hunk 4 above)
Comment (gabrielantao) [reply src/alfasim_score/conftest.py]: We are not running autoflake here 🫤 
Removed.

**Pull request merged**